src/github.com/matrix-org/dendrite/federationapi/writers/threepid.go

```diff
@@ -120,11 +121,17 @@ func createInviteFrom3PIDInvite(
 	}
 
 	if !queryRes.RoomExists {
-		// TODO: Use federation to auth the event
-		return nil, nil
+		// Use federation to auth the event
+		var remoteServer gomatrixserverlib.ServerName
+		_, remoteServer, err = gomatrixserverlib.SplitID('!', inv.RoomID)
```

Comment:
I think synapse also falls back to the server who sent the "m.room.third_party_invite" by using the user ID of the sender of the invite, or am I imagining things?

Reply:
Actually it first checks on the sender's server and then the room's server. Fixed in https://github.com/matrix-org/dendrite/pull/222/commits/0e356e2d272afae7640022d92c7ad68ca730e91a